### DISCUSSION TITLE
training using HF LORA - it trains only the transformer, should the text encode be trained as well?

### DISCUSSION THREAD

**Discussion by raulc0399** [PixArt-alpha/PixArt-alpha#133] [General]
checking the code for the [HF SDXL Lora script line 1034](https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/train_text_to_image_lora_sdxl.py#L1034)
i noticed that the text encoder is also trained.

any reason why is not trained for Pixart? i could try it, unless there is a good reason not to? :)

thanks.

**Comment by lawrence-cj** [comment]
We haven't tried to train the T5. You can give it a try. There is no specific reason for not training. Looking forward for your result~

**Reply by raulc0399** [reply]
have created the pull requst https://github.com/PixArt-alpha/PixArt-alpha/pull/138
i have tested it for syntax, it works for the different options.
i am still to test it for the full training, to see the effects, but i have created the PR just in case someone wants to try it already

**Reply by lawrence-cj** [reply]
Cool. Nice done!

**Reply by raulc0399** [reply]
added some prints to highlight that training in fp16 can lead to NaNs in loss. also some links with more info.

first tests on fp32 and bf16 however work, subjectively - i think the results are also slightly better :)

**Reply by raulc0399** [reply]
@lawrence-cj i have also added to the PR the option to train the text encoder at a different learning rate and also to stop training it at a percentage of the total steps

**Reply by raulc0399** [reply]
@lawrence-cj i have also added a PR for pixart-sigma with changes to train the text encoder

**Reply by lawrence-cj** [reply]
Cool cool. I will review it asap
